tools/cli/adminDBScanCommand.go

```diff
@@ -157,3 +163,137 @@ func checkExecution(
 
 	return execution, invariant.NewInvariantManager(ivs).RunChecks(ctx, execution)
 }
+
+func AdminDBScanUnsupportedWorkflow(c *cli.Context) {
+	numberOfShards := c.Int(FlagNumberOfShards)
+	rps := c.Int(FlagRPS)
+	outputFile := getOutputFile(c.String(FlagOutputFilename))
+	defer outputFile.Close()
+	for i := 0; i < numberOfShards; i++ {
+		listExecutionsByShardID(c, i, rps, outputFile)
+	}
+}
+
+func listExecutionsByShardID(
+	c *cli.Context,
+	shardID int,
+	rps int,
+	outputFile *os.File,
+) {
+
+	client := initializeExecutionStore(c, shardID, rps)
+	defer client.Close()
+
+	paginationFunc := func(paginationToken []byte) ([]interface{}, []byte, error) {
+		resp, err := client.ListConcreteExecutions(
+			context.Background(),
```

Comment:
not sure if this will cause the call to hang

Reply:
Updated with a timeout value